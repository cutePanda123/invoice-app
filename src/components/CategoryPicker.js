import React from 'react';
import PropTypes from 'prop-types';
import Ionicon from 'react-ionicons';

class CategoryPicker extends React.Component {
    constructor(props) {
        super(props);
    }

    selectCategory = (event, category) => {
        this.props.onSelectCategory(category);
        event.preventDefault();
    }

    render() {
        const { categories, selectedCategory } = this.props;
        const selectedCategoryId = selectedCategory && selectedCategory.id;
        return (
            <div className="category-picker-component">
                <div className="row">
                    {
                        categories.map((category, index) => {
                            const iconColor = (category.id === selectedCategoryId) ? '#fff' : '#555';
                            const backColor = (category.id === selectedCategoryId) ? '#347eff' : '#efefef';
                            const activeClassName = selectedCategoryId == category.id ? 
                                "category-item col-3 active" : "category-item col-3";
                            return (
                                <div
                                    className={activeClassName}
                                    key={index}
                                    onClick={(event) => {
                                        this.selectCategory(event, category);
                                    }}
                                    role='button'
                                    style={{textAlign: 'center'}}
                                >
                                    <Ionicon 
                                        className="rounded-circle"
                                        style={
                                            {
                                                backgroundColor: backColor,
                                                padding: '5px'
                                            }
                                        }
                                        fontSize="50px"
                                        color={iconColor}
                                        icon={category.iconName}
                                    />
                                    <p>{category.name}</p>
                                </div>
                            );
                        })
                    }
                </div>
            </div>
        );
    }
}

CategoryPicker.propTypes = {
    categories: PropTypes.array.isRequired,
    onSelectCategory: PropTypes.func.isRequired,
    selectedCategory: PropTypes.object
};

export default CategoryPicker;

