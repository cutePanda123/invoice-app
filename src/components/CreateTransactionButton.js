import React from 'react';
import Ionicon from 'react-ionicons';
import PropTypes from 'prop-types';

const CreateTransactionButton = ({ onCreateTransaction }) => {
    return (
        <button 
            className="btn btn-primary btn-block d-flex justify-content-center align-items-center"
            onClick={(e) => {
                onCreateTransaction()
            }}    
        >
            <Ionicon
                className="rounded-circle"
                fontSize="30px"
                color="#fff"
                icon="ios-add-circle"
            />
            Create a new transaction
        </button>
    );
}

CreateTransactionButton.propTypes = {
    onCreateTransaction: PropTypes.func.isRequired
}

CreateTransactionButton.defaultProps = {
    onCreateTransaction: () => {
        console.log("create a transaction button");
    }
}

export default CreateTransactionButton;