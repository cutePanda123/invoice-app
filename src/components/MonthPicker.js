import React from 'react';
import PropTypes from 'prop-types';
import Utility from '../utilility';

class MonthPicker extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
            selectedYear: props.year,
            selectedMonth: props.month
        };
    }

    onYearChange = (event, year) => {
        event.preventDefault();
        this.setState({
            selectedYear: year
        });
    }

    onMonthChange = (event, month) => {
        event.preventDefault();
        this.setState({
            selectedMonth: month,
            isOpen: false
        });
        this.props.onDateChange(this.state.selectedYear, this.state.selectedMonth);
    }


    onDropdownClick = (event) => {
        event.preventDefault();
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        const {selectedMonth, selectedYear} = this.state;
        const {year} = this.props;
        const { isOpen } = this.state;
        const monthRange = Utility.range(12, 1);
        const yearRange = Utility.range(9, -4).map(number => number + year);
        return (
            <div className="dropdown month-picker-component">
                <h4>Select a month</h4>
                <button
                    className="btn btn-lg btn-secondary dropdown-toggle"
                    onClick={this.onDropdownClick}
                >
                    {`${Utility.getMonthString(selectedMonth)}/${selectedYear}`}
                </button>
                {
                    isOpen &&
                    <div className="dropdown-menu" style={{display: 'block'}}>
                        <div className="row">
                            <div className="col border-right">
                                <a key="0" className="dropdown-item">Year</a>
                                {yearRange.map((curYear, idx) => {
                                    return (
                                        <a
                                            href="#"
                                            key={idx + 1} 
                                            className={curYear === selectedYear ? "dropdown-item active" : "dropdown-item"}
                                            onClick={
                                                (event) => {
                                                    this.onYearChange(event, curYear);
                                                }
                                            }
                                        >
                                            {curYear}
                                        </a>
                                    );
                                })}
                            </div>
                            <div className="col">
                                <a key="0" className="dropdown-item">Month</a>
                                {monthRange.map((curMonth, idx) => {
                                    return (
                                        <a
                                            href="#"
                                            key={idx + 1} 
                                            className={curMonth === selectedMonth ? "dropdown-item active" : "dropdown-item"}
                                            onClick={
                                                (event) => {
                                                    this.onMonthChange(event, curMonth);
                                                }
                                            }
                                        >
                                            {Utility.getMonthString(curMonth)}
                                        </a>
                                    );
                                })}
                            </div>
                        </div>
                    </div>
                }
            </div>
        );
    }
}

MonthPicker.propTypes = {
    year: PropTypes.number.isRequired,
    month: PropTypes.number.isRequired,
    onDateChange: PropTypes.func.isRequired
}

MonthPicker.defaultProps = {
    year: 1970,
    month: 1
}

export default MonthPicker;