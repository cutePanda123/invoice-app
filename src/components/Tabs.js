import React from 'react';
import PropTypes from 'prop-types';

export class Tabs extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            activeIndex: this.props.activeIndex
        };
    }

    clickTab = (event, index) => {
        event.preventDefault();
        this.setState({
            activeIndex: index
        });
        this.props.onTabChange(index);
    }

    render() {
        const { children } = this.props;
        const { activeIndex } = this.state;
        return (
            <ul className="nav nav-tabs nav-fill my-4">
                {
                    React.Children.map(children, (child, index) => {
                        const curClassName = activeIndex === index ? 'nav-link active' : 'nav-link';
                        return (
                            <li className="nav-item">
                                <a
                                    onClick={(event) => {this.clickTab(event, index);}}
                                    className={curClassName}
                                    href="#"
                                >
                                    {child}
                                </a>
                            </li>
                        );
                    })
                }
            </ul>
        );
    }
};

Tabs.propTypes = {
    activeIndex: PropTypes.number.isRequired,
    onTabChange: PropTypes.func.isRequired
};

export const Tab = ({children}) => {
    return <React.Fragment>{children}</React.Fragment>;
};