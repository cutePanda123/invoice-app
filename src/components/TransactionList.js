import React from 'react';

const TransactionList = ({ items, onModifyItem, onDeleteItem }) => {
    return (
        <ul className = "list-group list-group-flush">
            {
                items.map((item) => (
                    <li className = "list-group-item d-flex justify-content-between align-items-center"
                        key = {item.id}
                    >
                        <span className="col-1 badge badge-primary">
                            {item.category.name}
                        </span>
                        <span className="col-5">{item.desc}</span>
                        <span className="col-2 font-weight-bold">
                            {(item.category.type === 'income') ? '+' : '-'}
                            ${item.amount}
                        </span>
                        <span className="col-2">{item.date}</span>
                        <button 
                            className="col-1 btn btn-primary"
                            onClick={() => {
                                onModifyItem(item);
                            }}
                        >edit</button>
                        <button 
                            className="col-1 btn btn-danger"
                            onClick={() => {
                                onDeleteItem(item);
                            }}
                        >delete</button>
                    </li>
                ))
            }
        </ul>
    );
}

export default TransactionList;