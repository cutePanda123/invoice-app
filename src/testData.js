export const testCategories = [
  {
    "name": "travel",
    "iconName": "ios-plane",
    "id": "1",
    "type": "outcome"
  },
  {
    "name": "food",
    "iconName": "ios-restaurant",
    "id": "2",
    "type": "outcome"
  },
  {
    "name": "shopping",
    "iconName": "ios-basket",
    "id": "3",
    "type": "outcome"
  },
  {
    "name": "electronic product",
    "iconName": "ios-phone-portrait",
    "id": "4",
    "type": "outcome"
  },
  {
    "name": "工资",
    "iconName": "ios-card",
    "id": "10",
    "type": "income"
  },
  {
    "name": "part-time job",
    "iconName": "ios-cash",
    "id": "11",
    "type": "income"
  },
  {
    "name": "investment",
    "iconName": "logo-yen",
    "id": "12",
    "type": "income"
  }
];
  
export const testItems = [
  {
    "id": 0,
    "desc": "travel in Seattle",
    "amount": 2000,
    "date": "2020-10-20",
    "categoryId": 1
  },
  {
    "id": 1,
    "desc": "travel in San Jose",
    "amount": 4000,
    "date": "2020-12-10",
    "categoryId": 1
  },
  {
    "id": 2,
    "desc": "stock investement",
    "amount": 14000,
    "date": "2020-12-15",
    "categoryId": 2
  }
];

export const fakeTransaction = {
  "id": 0,
  "desc": "fake new transaction",
  "amount": 14000,
  "date": "2020-12-10",
  "categoryId": 2
};