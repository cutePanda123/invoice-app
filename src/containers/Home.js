import React from 'react';
import logo from '../logo.svg';
import TransactionList from '../components/TransactionList';
import Utility from '../utility';
import SpendingSummary from '../components/SpendingSummary';
import MonthPicker from '../components/MonthPicker';
import CreateTransactionButton from '../components/CreateTransactionButton';
import EditTransactionForm from '../components/EditTransactionForm';
import { Tabs, Tab } from '../components/Tabs';
import Ionicon from 'react-ionicons';
import { testItems, testCategories as categories, fakeTransaction } from '../testData';
import { AppContext } from '../App';

const tabTexts = [Utility.LIST_VIEW_NAME, Utility.CHART_VIEW_NAME];

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      items: testItems,
      currentDate: Utility.parseYearAndMonth(),
      tabView: Utility.LIST_VIEW_NAME
    };
  }

  changeView = (viewIndex) => {
    this.setState({
      tabView: tabTexts[viewIndex],
    });
  }

  changeDate = (year, month) => {
    this.setState({
      currentDate: {year, month}
    });
  }

  modifyTrasaction = (modifiedTransaction) => {
    const newTransactions = this.state.items.map((transaction) => {
      if (modifiedTransaction.id === transaction.id) {
        transaction.desc = "modified transaction";
      }
      return transaction;
    });
    this.setState({
      items: newTransactions
    });
  }

  createTransaction = () => {
    let newTrasaction = JSON.parse(JSON.stringify(fakeTransaction));
    newTrasaction.id = this.state.items.length === 0 ? 0 : Math.max.apply(Math, this.state.items.map(item => item.id)) + 1;
    const newItems = [newTrasaction, ...this.state.items];
    this.setState({
      items: newItems
    });
  }

  deleteTransaction = (victim) => {
    let transactions = this.state.items.filter(transaction => transaction.id !== victim.id);
    this.setState({
      items: transactions
    });
  }

  render() {
    const {items, currentDate, tabView} = this.state;
    const itemsWithCategory = items.map(item => {
      item.category = categories[item.categoryId];
      return item;
    }).filter((transaction) => {
      const month = Utility.getMonthString(currentDate.month);
      return transaction.date.includes(`${currentDate.year}-${month}`);
    });
    let totalIncome = 0, totalOutcome = 0;
    itemsWithCategory.forEach((item) => {
      if (item.category.type === Utility.OUTCOME_TYPE) {
        totalOutcome += item.amount;
      } else {
        totalIncome += item.amount;
      }
    });
    return (
      <AppContext.Consumer>
        {
          ({ state }) => {
            //console.log(state);

            return (
              <React.Fragment>
                <header className="App-header">
                  <div className="row mb-5">
                    <img src={logo} className="App-logo"></img>
                  </div>
                  <div className="row mb-5">
                    <EditTransactionForm
                      onFormCancel={() => {
                        console.log("on form cancle")
                      }}
                      onFormSubmit={() => {
                        console.log("on form submit")
                      }}
                      transaction={{}}
                    />
                  </div>
                  <div className="row mb-5">
                    <div className="col">
                      <MonthPicker
                        year={currentDate.year}
                        month={currentDate.month}
                        onDateChange={this.changeDate}
                      />
                    </div>
                    <div className="col">
                      <SpendingSummary
                        income={totalIncome}
                        outcome={totalOutcome}
                      />
                    </div>
                  </div>
                </header>
                <div className="cotent-area py-3 px-3">
                  <Tabs
                    activeIndex={0}
                    onTabChange={this.changeView}
                  >
                      <Tab>
                        <Ionicon
                          className="rounded-circle mr-2"
                          fontSize="25px"
                          color={"#007bff"}
                          icon="ios-paper"
                        />
                        List View
                      </Tab>
                      <Tab>
                        <Ionicon
                          className="rounded-circle mr-2"
                          fontSize="25px"
                          color={"#007bff"}
                          icon="ios-pie"
                        />
                        Chart View
                      </Tab>
                  </Tabs>
                  <CreateTransactionButton
                    onCreateTransaction={this.createTransaction}
                  />
                  { 
                    tabView === Utility.LIST_VIEW_NAME && 
                    <TransactionList 
                      items = {itemsWithCategory}
                      onDeleteItem = {this.deleteTransaction}
                      onModifyItem = {this.modifyTrasaction}
                    /> 
                  }
                  {
                    tabView === Utility.CHART_VIEW_NAME &&
                    <h2>Place holder for Chart View</h2>
                  }
                  
                </div>
              </React.Fragment>
            );
          }
        }
      </AppContext.Consumer>
    );
  }
}

export default Home;