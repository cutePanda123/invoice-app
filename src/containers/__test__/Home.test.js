import React from 'react';
import { mount } from 'enzyme';
import Home from '../Home';
import TransactionList from '../../components/TransactionList';
import { Tabs, Tab} from '../../components/Tabs';
import Utility from '../../utility';
import MonthPicker from '../../components/MonthPicker';
import CreateTransactionButton from '../../components/CreateTransactionButton';
import AppContext from '../../AppContext';
import { testItems, testCategories } from '../../testData'; 

let wrapper = null;

describe('Home test', () => {
    beforeEach(() => {
        wrapper = mount(
            <AppContext.Provider value={
                {
                    state: {
                        items: Utility.flattenArray(testItems),
                        categories: Utility.flattenArray(testCategories)
                    }
                }
            }>
                <Home />
            </AppContext.Provider>
        );
    });

    it('snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct dom', () => {
        const curDate = Utility.parseYearAndMonth('2020/12/31');
        expect(wrapper.find(TransactionList).length).toEqual(1);
        expect(wrapper.find(Tabs).props().activeIndex).toEqual(0);
        expect(wrapper.find(MonthPicker).props().year).toEqual(curDate.year);
        expect(wrapper.find(MonthPicker).props().month).toEqual(curDate.month);
        expect(wrapper.find(TransactionList).props().items).not.toBeNaN();
    });

    it('should switch tab when click tab', () => {
        wrapper.find('.nav-item a').first().simulate('click');
        expect(wrapper.find(Tabs).props().activeIndex).toEqual(0);
    });

    it('should change date tab when click a new month', () => {
        wrapper.find('.dropdown-toggle').simulate('click');
        wrapper.find('.months-range .dropdown-item').at(8).simulate('click');
        expect(wrapper.find(MonthPicker).props().month).toEqual(9);
    });

    it('should change date tab when click a new month', () => {
        wrapper.find('.dropdown-toggle').simulate('click');
        wrapper.find('.months-range .dropdown-item').at(8).simulate('click');
        expect(wrapper.find(MonthPicker).props().month).toEqual(9);
    });

    it('should create a transaction when click create button', () => {
        const oldTransactonNumber = wrapper.find(TransactionList).props().items.length;
        wrapper.find(CreateTransactionButton).simulate('click');
        expect(wrapper.find(TransactionList).props().items.length).toEqual(oldTransactonNumber + 1);
    });

    it('should delete a transaction when click delete button', () => {
        const oldTransactonNumber = wrapper.find(TransactionList).props().items.length;
        wrapper.find('.delete-button').first().simulate('click');
        expect(wrapper.find(TransactionList).props().items.length).toEqual(oldTransactonNumber - 1);
    });
});