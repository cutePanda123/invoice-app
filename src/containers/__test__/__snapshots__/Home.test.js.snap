// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test snapshot test 1`] = `
<Home>
  <header
    className="App-header"
  >
    <div
      className="row mb-5"
    >
      <img
        className="App-logo"
        src="logo.svg"
      />
    </div>
    <div
      className="row mb-5"
    >
      <CreateTransactionForm
        onFormCancel={[Function]}
        onFormSubmit={[Function]}
        transaction={Object {}}
      >
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="title"
            >
              Title
            </label>
            <input
              aria-describedby="titleHelp"
              className="form-control"
              id="title"
              placeholder="Enter title"
              type="text"
            />
            <small
              className="form-text text-muted"
              id="titleHelp"
            >
              Title for your transaction.
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="amount"
            >
              Dollar Amount
            </label>
            <input
              className="form-control"
              id="amount"
              placeholder="Dollar Amount"
              type="number"
            />
            <small
              className="form-text text-muted"
              id="amountHelp"
            >
              Dollar amount for your transaction.
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="date"
            >
              Date
            </label>
            <input
              className="form-control"
              id="date"
              placeholder="Date"
              type="date"
            />
            <small
              className="form-text text-muted"
              id="dateHelp"
            >
              Date for your transaction.
            </small>
          </div>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
          <button
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Cancel
          </button>
        </form>
      </CreateTransactionForm>
    </div>
    <div
      className="row mb-5"
    >
      <div
        className="col"
      >
        <MonthPicker
          month={12}
          onDateChange={[Function]}
          year={2020}
        >
          <div
            className="dropdown month-picker-component"
          >
            <h4>
              Select a month
            </h4>
            <button
              className="btn btn-lg btn-secondary dropdown-toggle"
              onClick={[Function]}
            >
              12/2020
            </button>
          </div>
        </MonthPicker>
      </div>
      <div
        className="col"
      >
        <SpendingSummary
          income={0}
          outcome={18000}
        >
          <div
            className="containers"
          >
            <div
              className="row"
            >
              <div
                className="col income"
              >
                <h5>
                  Income: 
                </h5>
                <div>
                  0
                </div>
              </div>
              <div
                className="col outcome"
              >
                <h5>
                  Outcome: 
                </h5>
                <div>
                  18000
                </div>
              </div>
            </div>
          </div>
        </SpendingSummary>
      </div>
    </div>
  </header>
  <div
    className="cotent-area py-3 px-3"
  >
    <Tabs
      activeIndex={0}
      onTabChange={[Function]}
    >
      <ul
        className="nav nav-tabs nav-fill my-4"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link active"
            href="#"
            onClick={[Function]}
          >
            <Tab>
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              >
                <styled.svg
                  beat={0}
                  className="rounded-circle mr-2"
                  fill="#007bff"
                  height="25px"
                  rotate={0}
                  shake={0}
                  style={Object {}}
                  viewBox="0 0 1024 1024"
                  width="25px"
                >
                  <svg
                    className="rounded-circle mr-2 sc-bdVaJa fUuvxv"
                    fill="#007bff"
                    height="25px"
                    rotate={0}
                    style={Object {}}
                    viewBox="0 0 1024 1024"
                    width="25px"
                  >
                    <path
                      d="M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z"
                    />
                  </svg>
                </styled.svg>
              </Ionicon>
              List View
            </Tab>
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link"
            href="#"
            onClick={[Function]}
          >
            <Tab>
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              >
                <styled.svg
                  beat={0}
                  className="rounded-circle mr-2"
                  fill="#007bff"
                  height="25px"
                  rotate={0}
                  shake={0}
                  style={Object {}}
                  viewBox="0 0 1024 1024"
                  width="25px"
                >
                  <svg
                    className="rounded-circle mr-2 sc-bdVaJa fUuvxv"
                    fill="#007bff"
                    height="25px"
                    rotate={0}
                    style={Object {}}
                    viewBox="0 0 1024 1024"
                    width="25px"
                  >
                    <path
                      d="M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z"
                    />
                  </svg>
                </styled.svg>
              </Ionicon>
              Chart View
            </Tab>
          </a>
        </li>
      </ul>
    </Tabs>
    <CreateTransactionButton
      onCreateTransaction={[Function]}
    >
      <button
        className="btn btn-primary btn-block d-flex justify-content-center align-items-center"
        onClick={[Function]}
      >
        <Ionicon
          beat={false}
          className="rounded-circle"
          color="#fff"
          fontSize="30px"
          icon="ios-add-circle"
          rotate={false}
          shake={false}
          style={Object {}}
        >
          <styled.svg
            beat={0}
            className="rounded-circle"
            fill="#fff"
            height="30px"
            rotate={0}
            shake={0}
            style={Object {}}
            viewBox="0 0 1024 1024"
            width="30px"
          >
            <svg
              className="rounded-circle sc-bdVaJa fUuvxv"
              fill="#fff"
              height="30px"
              rotate={0}
              style={Object {}}
              viewBox="0 0 1024 1024"
              width="30px"
            >
              <path
                d="M512 96c-229.8 0-416 186.2-416 416s186.2 416 416 416c229.8 0 416-186.2 416-416s-186.2-416-416-416zM768 528h-240v240h-32v-240h-240v-32h240v-240h32v240h240v32z"
              />
            </svg>
          </styled.svg>
        </Ionicon>
        Create a new transaction
      </button>
    </CreateTransactionButton>
    <TransactionList
      items={
        Array [
          Object {
            "amount": 4000,
            "category": Object {
              "iconName": "ios-restaurant",
              "id": "2",
              "name": "food",
              "type": "outcome",
            },
            "categoryId": 1,
            "date": "2020-12-10",
            "desc": "travel in San Jose",
            "id": 1,
          },
          Object {
            "amount": 14000,
            "category": Object {
              "iconName": "ios-basket",
              "id": "3",
              "name": "shopping",
              "type": "outcome",
            },
            "categoryId": 2,
            "date": "2020-12-15",
            "desc": "stock investement",
            "id": 2,
          },
        ]
      }
      onDeleteItem={[Function]}
      onModifyItem={[Function]}
    >
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
          key="1"
        >
          <span
            className="col-1"
          >
            <Ionicon
              beat={false}
              className="rounded-circle"
              color="#fff"
              fontSize="30px"
              icon="ios-restaurant"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#007bff",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M768 128c-49.6 0-96 154.6-96 320 0 114 64 160 64 160v256c0 17.6 14.4 32 32 32s32-14.4 32-32v-714c0-22-22-22-32-22z M576 128l20 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l-12-208h-16l-12 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l20-208h-16c0 0-48 214.4-48 256s26.8 77.2 64 90.4v389.6c0 17.6 14.4 32 32 32s32-14.4 32-32v-389.6c37.2-13.2 64-48.4 64-90.4s-48-256-48-256h-16z M320 128c-53 0-96 128-96 256 0 41.6 26.8 77.2 64 90.4v389.6c0 17.6 14.4 32 32 32s32-14.4 32-32v-389.6c37.2-13.2 64-48.4 64-90.4 0-128-43-256-96-256z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </span>
          <span
            className="col-5"
          >
            travel in San Jose
          </span>
          <span
            className="col-2 font-weight-bold"
          >
            -
            $
            4000
          </span>
          <span
            className="col-2"
          >
            2020-12-10
          </span>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle"
              color="#fff"
              fontSize="30px"
              icon="ios-create-outline"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M736 832h-576v-544h448.234l32-32h-512.234v608h640v-480.234l-32 32z M475.246 568.168l-19.412-19.414 342.646-342.646-22.626-22.628-359.854 359.822v64.698h64.432l360.090-359.968-22.628-22.572z M888.426 160.624l-24.93-24.932c-5.328-4.758-12.384-7.692-20.004-7.692-7.624 0-14.608 2.936-19.858 7.7l-24.302 24.3 64.786 64.786 24.3-24.3c4.766-5.25 7.582-12.234 7.582-19.858 0-7.62-2.816-14.678-7.574-20.004z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </a>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle delete-button"
              color="#fff"
              fontSize="30px"
              icon="ios-close"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle delete-button"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle delete-button sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M680.4 320l-168.8 168.6-168-167.8-23.6 23.6 168 167.6-168 167.8 23.6 23.4 168-167.6 168.8 168.4 23.6-23.4-168.8-168.6 168.8-168.4z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </a>
        </li>
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
          key="2"
        >
          <span
            className="col-1"
          >
            <Ionicon
              beat={false}
              className="rounded-circle"
              color="#fff"
              fontSize="30px"
              icon="ios-basket"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#007bff",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M800 448v-192c0-70.4-57.6-128-128-128h-320c-70.4 0-128 57.6-128 128v192h-160l105 388.6c9.4 34 40.8 59.4 78.2 59.4h529.4c37.4 0 68.8-25 78.6-59l104.8-389h-160zM256 256c0-25.6 10-49.6 28.2-67.8s42.2-28.2 67.8-28.2h320c25.6 0 49.6 10 67.8 28.2s28.2 42.2 28.2 67.8v192h-512v-192zM883.8 608h-147.8v128h113.4l-8.6 32h-104.8v96h-32v-96h-176v96h-32v-96h-176v96h-32v-96h-104.4l-8.6-32h113v-128h-147.6l-8.6-32h156.2v-96h32v96h176v-96h32v96h176v-96h32v96h156.4l-8.6 32z M528 608h176v128h-176v-128z M320 608h176v128h-176v-128z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </span>
          <span
            className="col-5"
          >
            stock investement
          </span>
          <span
            className="col-2 font-weight-bold"
          >
            -
            $
            14000
          </span>
          <span
            className="col-2"
          >
            2020-12-15
          </span>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle"
              color="#fff"
              fontSize="30px"
              icon="ios-create-outline"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M736 832h-576v-544h448.234l32-32h-512.234v608h640v-480.234l-32 32z M475.246 568.168l-19.412-19.414 342.646-342.646-22.626-22.628-359.854 359.822v64.698h64.432l360.090-359.968-22.628-22.572z M888.426 160.624l-24.93-24.932c-5.328-4.758-12.384-7.692-20.004-7.692-7.624 0-14.608 2.936-19.858 7.7l-24.302 24.3 64.786 64.786 24.3-24.3c4.766-5.25 7.582-12.234 7.582-19.858 0-7.62-2.816-14.678-7.574-20.004z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </a>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle delete-button"
              color="#fff"
              fontSize="30px"
              icon="ios-close"
              rotate={false}
              shake={false}
              style={
                Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                }
              }
            >
              <styled.svg
                beat={0}
                className="rounded-circle delete-button"
                fill="#fff"
                height="30px"
                rotate={0}
                shake={0}
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                viewBox="0 0 1024 1024"
                width="30px"
              >
                <svg
                  className="rounded-circle delete-button sc-bdVaJa fUuvxv"
                  fill="#fff"
                  height="30px"
                  rotate={0}
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="30px"
                >
                  <path
                    d="M680.4 320l-168.8 168.6-168-167.8-23.6 23.6 168 167.6-168 167.8 23.6 23.4 168-167.6 168.8 168.4 23.6-23.4-168.8-168.6 168.8-168.4z"
                  />
                </svg>
              </styled.svg>
            </Ionicon>
          </a>
        </li>
      </ul>
    </TransactionList>
  </div>
</Home>
`;
